### DISCUSSION TITLE
AWS-Cloudwatch logs not been shown on dashboard

### DISCUSSION THREAD

**Discussion by usmanaa10** [wazuh/wazuh#13917] [General]
Hi Friends,
               I have configured aws cloudwatch and didnt see logs on wazuh dashboard..
1) getting no error in ossec.log
2) run wodle manually and got the following output
![cloudwatch](https://user-images.githubusercontent.com/66419518/174558476-3eb8717b-b5e0-4a8b-a9c2-a1e1fb1be6da.png)

Can  anyone take out of this problem. ASAP.. Please?

Regards

**Comment by Selutario** [comment]
Hi @usmanaa10,

As you say, there are no errors in the `ossec.log` and the screenshot shows that information is being obtained from AWS. This means that there is data but there are (probably) no decoders/rules to parse it. 

Remember the way Wazuh works:
- Managers receive logs (also called events). 
- The logs are decoded and compared to rules. 
- If the log matches any rule, an alert is generated. 
- Alerts (only alerts, no logs) are sent and displayed on the Wazuh Dashboard.

Please note that since you can process any log with CloudWatch Logs, this prevents us from creating predefined decoders and rules in Wazuh, which means you must make your own.

First, you should check what logs are coming to Wazuh to create your decoders and rule. To do this, you will need to enable the [logall_json](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) option within your manager's `ossec.conf`. After enabling it and restarting the Wazuh manager, all events that reach your manager will appear inside `/var/ossec/logs/archives/archives.json`, whether they generate an alert or not.

There you should look for the events that are captured by your Cloudwatch logs integration. Copy one that you find interesting (the entire line). Now, you will have to run this binary and paste the line inside to determine if the log is being captured by any decoder and rule:
```
/var/ossec/bin/wazuh-logtest
```

The last step will be to create your own decoders and rules so that the captured logs become alerts. To do this, follow the instructions on this documentation page:
- [Custom rules and decoders - Wazuh](https://documentation.wazuh.com/current/user-manual/ruleset/custom.html)

---------
**P.S:** Remember to disable [logall_json](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#logall-json) and restart Wazuh again as soon as you have copied the Cloudwatch logs you are interested in. Otherwise, a lot of space on your disk could be used to store all the received events.

**Comment by usmanaa10** [comment]
Hi @Selutario 
              Thanks for reply and and thank for explain briefly
What I did
 I have enable all logs, and tail the  tail -f /var/ossec/logs/archives/archives.log | grep aws
but I got
![image](https://user-images.githubusercontent.com/66419518/175002676-5c0b2fff-6ca3-4f54-ac64-d82886271d60.png)

While running wodle manually i got no error
using
following command
![image](https://user-images.githubusercontent.com/66419518/175002735-603e2788-bc74-4fa1-8936-5dddbbc20cfb.png)

results are
![image](https://user-images.githubusercontent.com/66419518/175002781-e7ee359e-0bb7-44c7-a04a-85d4bb24abfd.png)


![image](https://user-images.githubusercontent.com/66419518/175002487-e7f12828-5af0-4795-814e-9550afad20ad.png)


Can you help me out where i am doing mistake , Actually i give access to wazuh user to access cloudwatch logs

Note : I have configure aws-waf but its work properly

**Reply by Selutario** [reply]
Hi usmanaa10,

Events created by the Cloudwatch logs integration have this header: `Wazuh-AWS`. Notice that AWS is in uppercase, so the grep you've used won't be able to find it. I recommend you using `-i` to make it case insensitive:
```
tail -f /var/ossec/logs/archives/archives.log | grep -i aws
```

**Reply by usmanaa10** [reply]
Hi @Selutario 
                 I use same command as u mention above but still i got this problem
![image](https://user-images.githubusercontent.com/66419518/175265614-b329c475-6ee2-4157-a8f3-37234ec6f420.png)

while running wodle manually i didn't face any error .

Any other way to sort out

**Reply by Selutario** [reply]
Hi usmanaa10,

There is a possibility that the log streams that Wazuh is reading in CloudWatch logs are empty. If it does find events, each of them will be sent to Analysisd (the Wazuh service where they are decoded and matched with rules to generate alerts). When sending them, the module should print (in debug mode) the following message for each of the events sent:
```
+++ Sending events to Analysisd...
```

I see that this message does not appear in your screenshots, which could mean that they were indeed empty. Regarding this, there are some things that you should take into account or verify:
1. A log stream could have data in one region (for example `us-east-1`) and be empty in another. Make sure the region you are specifying in the Wazuh wodle contains data.
2. You are not specifying the [only_logs_after](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/wodle-s3.html#service-only-logs-after) option. It means that only logs from the current date (today) will be parsed. Therefore, before running the wodle, make sure that the log stream contains logs for today (in the specified region).
3. I'm not entirely sure about this, but it could happen that you don't have enough permissions to get those logs without causing the Wazuh module to fail. Verify that the authentication method it uses contains the following permissions:
  ```JSON
    {
	"Version": "2012-10-17",
	"Statement": [
	    {
		"Sid": "VisualEditor1",
		"Effect": "Allow",
		"Action": [
              "logs:DescribeLogStreams",
		    "logs:GetLogEvents",
		    "logs:DeleteLogStream"
		],
		"Resource": "*"
	    }
	]
    }
  ```

**Comment by Truongvt2** [comment]
hello @Selutario I got the same issue, that is aws-cloudwatch logs not been shown on wazuh-dashboard
(I follow this guide: https://github.com/wazuh/wazuh-docker.git -b v4.4.1, with single-model)
Here is log from archives.log when I run command: tail -f archives/archives.log | grep -i aws

![error3](https://user-images.githubusercontent.com/32538408/233907190-0ef3f400-924a-444c-9ca7-54f82a724c8a.jpg)

Error Message: 
any->aws-s3 DEBUG: The message is "2023-04-24 05:11:56 INFO [func1 @ sendreply.go.77] [ssm-agent-worker] [MessageService] [MGSInteractor] send failed reply thread done
and
any->aws-s3 sqlite3.OperationalError: database or disk is full
then I see this error from archiver.json
But I checked I dont see full disk in docker container or host, please kindly help to debug

**Reply by Selutario** [reply]
Hi Truongvt2,

It doesn't seem that this error is the same as the one in the original question, so it would be advisable to open a new thread to avoid mixing topics. In any case, I've been trying to reproduce it without success. 

The database created by the AWS integration in Wazuh is used only to determine which logs have already been processed and which have not. If you delete it, it will be automatically recreated the next time the integration is run. In case the problem is not really due to full disk, this might fix it. 

Note however that it could happen that events from cloudwatch and other integrations that have already been processed in the past are reprocessed, generating duplicate alerts. If you don't use the [<only_logs_after>](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/wodle-s3.html#only-logs-after) option in the configuration, only logs after the current day will be processed/reprocessed.

Instead of deleting the DB, I recommend you to rename it as backup. You can do it like this:
```
mv /var/ossec/wodles/aws/aws_services.db /var/ossec/wodles/aws/backup_aws_services.db
```

**Reply by Truongvt2** [reply]
hello @Selutario  I already create new thread https://github.com/wazuh/wazuh/discussions/16921
Please kindly help to check
